### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,88 @@
...
+        <ul>
+          {goods.map(good => (
+            <li
+              style={{ color: good.color }}
+            >
+              {good.name}
+            </li>
+          ))}
+        </ul>
+      )}
```
Comment (Andriyk0) [inline src/App.tsx:83]: 1. Create a component `GoodsList` accepting an array of goods and rendering them inside a `<ul>`

**Review by Andriyk0 — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,81 @@
...
-// import * as goodsAPI from './api/goods';
+import { getAll } from './api/goods';
+import GoodsList from './GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState([{ id: 0, name: '', color: '' }]);
+  const [listIsVisible, setListIsVisible] = useState(false);
```
Comment (Anna-Tkachenko) [inline src/App.tsx:10]: this variable is redundant, just show an empty array as initial array

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/api/goods.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
+};
+
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
     .then(response => response.json());
 }
-
-export const get5First = () => {};
-
-export const getRedGoods = () => {};
```
Comment (Anna-Tkachenko) [inline src/api/goods.ts:11]: Don't delete this functions, you should use them in your code
On '5' button you should call get5First function, on 'red' button you should call getRedGoods function

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/api/goods.ts (hunk 3 above)
Comment (yelnikov-andrii) [reply src/api/goods.ts:11]: but these functions are the same

File: src/api/goods.ts [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
     .then(response => response.json());
 }
 
-export const get5First = () => {};
+export const get5First = () => {
+  return getAll()
+    .then(theGoods => {
+      const five = theGoods.filter(good => {
+        return good.id <= 5;
+      });
```
Comment (Anna-Tkachenko) [inline src/api/goods.ts:20]: You forget about sorting
Load 5 first goods shows 5 first goods after sorting them by name

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -1,12 +1,65 @@
...
+  };
+
+  return (
+    <>
+      <h1>Dynamic list of Goods</h1>
+      <button
+        type="button"
+        onClick={() => {
+          showAll();
+        }}
```
Comment (Denchepornyuk) [inline src/App.tsx:37]:
```suggestion
        onClick={showAll}
```
fix in all places

File: src/GoodsList.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  goods: Good[]
+};
+
+const GoodsList: React.FC <Props> = ({ goods }) => {
+  return (
+    <ul className="list">
+      {goods.map(good => (
+        <li
+          style={{ color: good.color }}
+        >
```
Comment (Denchepornyuk) [inline src/GoodsList.tsx:17]: add key attribute

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**